presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -421,6 +422,11 @@ public void addFunctionSignatures(IdentityLinkedHashMap<FunctionCall, Signature>
         return unmodifiableSet(columnReferences.keySet());
     }
 
+    public Map<Expression, FieldId> getColumnReferenceFields()
```

Comment:
Having bindings of column references is great benefit of this PR. I am planning to use bindings in `AggregationAnalyzer` and `TranslationMap` in order to compare expressions and references. With bindings you can correctly compare e.g: `t.a+b` with `a+b` where `t.a` and `a` are references to the same field. You can also distinguish (e.g: in `TranslationMap`) between `a` (output column reference) and `a` (`FROM` column reference).
Bindings allow you to replace `rewriteBase` and `fieldSymbols` in `TranslationMap` with just `Map<FieldId, Symbol> fieldMappings`. This will simplify `TranslationMap` and `QueryPlanner` since you don't have to depend on `Scopes` anymore.

Now, will we eventually need `getColumnReferences`? That depends how we would like to compare AST expressions. If we compare them with visitors (that traverses both operands at the same time) then `Analysis` and `getColumnReferences` will be required. If we replace `Identifiers` and `DereferenceExpression` with `FieldReference` (referencing `FieldId`) in expressions and keep them in `Analysis` then we could remove `getColumnReferences` method. The latter additionally won't require dedicated expression comparison visitor.

See my proposal: https://github.com/prestodb/presto/issues/7398